Feature/create account form validation

xErikPsx opened the pull request:
# Description
In this PR were created all the validators for each case of wrong input password and a wrong email, each case gets highlighted in color red if a term is missing and the return to the original color.

#Screenshots

![image](https://user-images.githubusercontent.com/26925761/97641567-d0331000-1a08-11eb-9ee6-728433b36968.png)

closes #5

xErikPsx committed:
Set up the intial state

xErikPsx committed:
Creating onHandleChange function

xErikPsx committed:
Passing value prop to chilren inputs

xErikPsx committed:
Correcting eslint issues

xErikPsx committed:
Create function IsEmptyInput to validate each filed filled

xErikPsx committed:
Set up a color state for invalid data

xErikPsx committed:
Create validateEmail function with its regex

xErikPsx committed:
Create validatePassword function with its regex

xErikPsx committed:
Passing onChangeText props

xErikPsx committed:
Create validator function's file

xErikPsx committed:
Setting the color changing for review

cfcortes commented:
@xErikPsx on PRs Description you must specify or explain the purpose of the intention of your contibution (coder), this is why or what you have implemented and submitted.

I understand that in this case, you submitted your PR so others could have a look at your code and help you. Still should have described your PR, and perhaps you could have added a note or an additional note asking for help t other (mentors).

I think you can still edit your description. I so, please add it.

xErikPsx commented:
I just tried to upload the PR as soon as I could because Aramando was answering me quickly and I wanted to him checked asap because I know their busy too. But of course, I will edit my PR review.

luisarmandodv commented on src/screens/RegisterAccount.js:
```diff
@@ -1,43 +1,169 @@
+              Alert.alert('Las contraseñas no coinciden');
+              return false;
+            }
+          } else {|
```
```suggestion
          } else {
```

luisarmandodv commented on src/screens/RegisterAccount.js:
```diff
@@ -1,43 +1,169 @@
+          setColor({...color, passwordInput: false, thirdTerm: false});
+          console.log('>= 8 y minuscula y mayuscula');
+          if (arePasswordsCorrect.number) {
+            setColor({...color, passwordInput: false, fourthdTerm: false});
```
```suggestion
            setColor({...color, passwordInput: false, fourthTerm: false});
```

luisarmandodv commented on src/screens/RegisterAccount.js:
```diff
@@ -1,43 +1,169 @@
+      data.validatePassword,
+    );
+    if (arePasswordsCorrect.lengthPassword) {
+      console.log('>= 8');
+      setColor({...color, passwordInput: false, firsTerm: false});
+      if (arePasswordsCorrect.lowerCase) {
+        setColor({...color, passwordInput: false, secondTerm: false});
+
+        console.log('>= 8 y minuscula');
+        if (arePasswordsCorrect.caps) {
+          setColor({...color, passwordInput: false, thirdTerm: false});
+          console.log('>= 8 y minuscula y mayuscula');
+          if (arePasswordsCorrect.number) {
+            setColor({...color, passwordInput: false, fourthdTerm: false});
+            console.log('>= 8 y minuscula y mayuscula y numero');
+            if (data.password === data.validatePassword) {
+              setColor({...color, passwordInput: false});
+              return true;
+            } else {
+              setColor({...color, passwordInput: true});
+              Alert.alert('Las contraseñas no coinciden');
+              return false;
+            }
+          } else {|
+            setColor({...color, passwordInput: true, fourthTerm: true});
+            console.log('< 8, no minuscula, no mayusc, no num');
+          }
+        } else {
+          setColor({...color, passwordInput: true, thirdTerm: true});
+          console.log('< 8, no minuscula, no mayusc');
+        }
+      } else {
+        setColor({...color, passwordInput: true, secondTerm: true});
+        console.log('< 8 no minuscula');
+      }
+    } else {
+      setColor({...color, passwordInput: true, firstTerm: true});
+      console.log(' es < 8');
+    }
+  };
```
```suggestion
  const validateOnPasswords = () => {
    const { lengthPassword, lowerCase, caps, number } = validatePasswords(
      data.password,
      data.validatePassword,
    );

    const passwordInput = !lengthPassword || !lowerCase || !caps || !number || !(data.password === data.validatePassword);
    
    setColor({
      ...color,
      passwordInput,
      firsTerm: !lengthPassword,
      secondTerm: !lowerCase,
      thirdTerm: !caps,
      fourthTerm: !number
    });

    return passwordInput;
  };
```

xErikPsx committed:
Update the regex using constructors (RegExp)

xErikPsx committed:
Update component's colors

xErikPsx committed:
Fix input validators' state

OwenRamire committed:
Merge branch 'sand-box' into feature/create-account-form-validation

xErikPsx committed:
Fixing prettier issues

xErikPsx committed:
Merge branch 'feature/create-account-form-validation' of https://github.com/BrightCoders-Proyectos/covid-tracker into feature/create-account-form-validation

xErikPsx committed:
Fix ESlint issues

OwenRamire reviewed (dismissed):
🚀

luisarmandodv commented on src/components/PasswordRequirements.js:
```diff
@@ -34,6 +40,7 @@ const styles = StyleSheet.create({
   },
   textInfo: {
     fontSize: 15,
+    // color: Colors.red
```
```suggestion
```

luisarmandodv commented on src/components/PasswordRequirements.js:
```diff
@@ -1,22 +1,28 @@
+import {Colors} from '../utils/Colors';
 
-const PasswordRequirements = () => {
+const PasswordRequirements = (props) => {
```
```suggestion
const PasswordRequirements = (props) => {
  const keyItemsArray = ['firstTerm', 'secondTerm', 'thirdTerm', 'fourthTerm'];

  return keyItemsArray.map( key => {
    const color = props[key] ? Colors.red : Colors.black;
    const text = PASS_REQUIREMENTS[key];
    return (
      <Text style={[styles.textInfo, {color}]} key={key}>
        <Text style={styles.dash}>- </Text>
        {text}
      </Text>
      )
    )
  };
```

luisarmandodv replied:
All that function, can be refactored to this

luisarmandodv commented on src/screens/RegisterAccount.js:
```diff
@@ -8,38 +8,135 @@ import {
+    const isUserPasswordValidation = validateOnPasswords();
+    if (isEmailEmpty && isPasswordEmpty && isValidatePassEmpty) {
+      if (validateOnEmail() === true && isUserPasswordValidation === false) {
+        console.log('Exito');
```
```suggestion
```

luisarmandodv commented on src/screens/RegisterAccount.js:
```diff
@@ -8,38 +8,135 @@ import {
+      data.password,
+      data.validatePassword,
+    );
+    console.log('validateOnPasswords -> lengthPassword', lengthPassword);
```
```suggestion
```

luisarmandodv reviewed (changes requested):
Remember to use PropTypes on ALL the components

https://es.reactjs.org/docs/typechecking-with-proptypes.html

fernandoamz commented on src/screens/RegisterAccount.js:
```diff
@@ -8,38 +8,135 @@ import {
+          usersEmail: data.email,
+        });
+      } else {
+        console.log('Sin exito');
```
```suggestion
return
```

fernandoamz commented on src/screens/RegisterAccount.js:
```diff
@@ -8,38 +8,135 @@ import {
+            id="email"
+            keyboard="email-address"
+            value={data.email}
             title="Correo electrónico"
```
Add to constant file

fernandoamz commented on src/screens/RegisterAccount.js:
```diff
@@ -8,38 +8,135 @@ import {
+            id="password"
+            keyboard="default"
+            value={data.password}
             title="Clave"
```
Add to constant file

LeyaimMagma commented on src/components/PasswordRequirements.js:
```diff
@@ -1,22 +1,28 @@
 
-const PasswordRequirements = () => {
+const PasswordRequirements = (props) => {
+  const firstTermWrong = props.firstTerm ? Colors.red : Colors.black;
+  const secondTermWrong = props.secondTerm ? Colors.red : Colors.black;
+  const thirdTermWrong = props.thirdTerm ? Colors.red : Colors.black;
+  const fourthTermWrong = props.fourthTerm ? Colors.red : Colors.black;
```
You can refactor this and create a function

OwenRamire committed:
Change state on the EmailInput

For a better UX, if the user touch the button without type his email and
password, the inputs' borders are going to change their color to red.
So, after that all the fields will be red and the user will type them on
that color. With this change if the user types a validate email but a
wrong password, the email's field will return at his original color and
the password's field will keep in red color.
Also I delete unnecessary comments

OwenRamire committed:
Refactor passwordRequirements component

OwenRamire committed:
Use PropTypes

OwenRamire committed:
Merge branch 'sand-box' into feature/create-account-form-validation
